src/Bicep.Core/Emit/EmitLimitationCalculator.cs

```diff
@@ -81,7 +81,7 @@ private static IEnumerable<ModuleDefinition> GetModuleDefinitions(SemanticModel
 
         private static IEnumerable<ResourceDefinition> GetResourceDefinitions(SemanticModel semanticModel, ImmutableDictionary<ResourceSymbol, ScopeHelper.ScopeData> resourceScopeData)
         {
-            foreach (var resource in semanticModel.Root.ResourceDeclarations)
+            foreach (var resource in semanticModel.Root.GetAllResourceDeclarations())
```

Comment:
Now that resources can appear at multiple levels, you need to consume them differently. `Root.ResourceDeclarations` only contains the immediate children (top level).

Reply:
Makes sense.

---
In reply to: [577214441](https://github.com/Azure/bicep/pull/1516#discussion_r577214441) [](ancestors = 577214441)